Adafruit_MQTT.cpp

```diff
@@ -96,6 +96,22 @@ static uint8_t *stringprint(uint8_t *p, const char *s, uint16_t maxlen = 0) {
   return p + len;
 }
 
+// packetAdditionalLen is a helper function used to figure out
+// how bigger the payload needs to be in order to account for
+// its variable length field. As per
+// http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Table_2.4_Size
+// See also readFullPacket
+static uint16_t packetAdditionalLen(uint32_t currLen) {
```

Comment:
I changed the signature here to explicitly take a uint32_t so that the comparisons are consistent. This doesn't change the fact that the max packet length is limited to 65536, but that is checked in readFullPacket